Readme.md

```diff
@@ -108,37 +108,92 @@ See the Concurrent Updates section for details how to implement this in a custom
 ## Type Marshaling
 
 `ddbus` includes fancy templated methods for marshaling D types in and out of DBus messages.
-All DBus-compatible basic types work (except dbus path objects and file descriptors).
-Any forward range can be marshalled in as DBus array of that type but arrays must be taken out as dynamic arrays.
-Structures are mapped to `Tuple` from `std.typecons`.
+All DBus-compatible basic types work (except file descriptors).
+Any forward range can be marshaled in as DBus array of that type but arrays must be taken out as dynamic arrays.
+
+As per version 2.3.0, D `struct` types are fully supported by `ddbus`. By default all public fields of a structure are marshaled. This behavior can be changed by UDAs. Mapping DBus structures to a matching instance of `std.typecons.Tuple`, like earlier versions of `ddbus` did, is also still supported.
 
 Example using the lower level interface, the simple interfaces use these behind the scenes:
 ```d
 Message msg = Message("org.example.wow","/wut","org.test.iface","meth");
-bool[] emptyB;
-auto args = tuple(5,true,"wow",[6,5],tuple(6.2,4,[["lol"]],emptyB));
+
+struct S {
+  double a;
+  int b;
+  string[][] c;
+  bool[] d;
+}
+
+auto s = S(6.2, 4, [["lol"]], []);
+auto args = tuple(5, true, "wow", [6, 5], s);
 msg.build(args.expand);
 msg.signature().assertEqual("ibsai(diaasab)");
 msg.readTuple!(typeof(args))().assertEqual(args);
 ```
+### Basic types
+These are the basic types supported by `ddbus`:
+`bool`, `byte`, `short`, `ushort`, `int`, `uint`, `long`, `ulong`, `double`, `string`, `ObjectPath`
+
+### Overview of mappings of other types:
+
+| D type | DBus type | Comments |
+| --- | --- | --- |
+| any `enum` | `enum` base type | Only the exact values present in the definition of the `enum` type will be allowed. |
+| `std.typecons.BitFlags` | `enum` base type | Allows usage of OR'ed values of a flags `enum`. |
+| dynamic array `T[]` | array | |
+| associative array `V[K]` | array of key-value pairs | DBus has a special type for key-value pairs, which can be used as the element type of an array only. |
+| `Tuple!(T...)` | structure | The DBus structure will map all of the `Tuple`'s values in sequence. |
+| any `struct` | structure | The DBus structure will map all public fields of the `struct` type in order of definition, unless otherwise specified using UDAs. |
+| `ddbus` style variant `Variant!T` | variant | `Variant!T` is in fact just a wrapper type to force representation as a variant in DBus, use `Variant!DBusAny` for actual dynamic typing. |
+| Phobos style variants `std.variant.VariantN` | variant | Only supported if set of allowed types is limited to types that can be marshaled by `ddbus`, so `std.variant.Variant` is not supported, but `std.variant.Algebraic` may be, depending on allowed types |
```

Comment:
Uh I know this will be nicely formatted by github but for people who cloned the repository and now look at the README in vim or some other text editor this will be really unreadable, can you add spaces in the entire table to make it look like a table in text mode